### PR TITLE
HoH heartbeat crd

### PR DESCRIPTION
**Pull request by OlegSternbergIBM**
Added HoH heartbeat CRD which will be used in leaf hub to track HoH availability

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,22 +43,26 @@ controller-gen: ## Download controller-gen locally if necessary.
...
 ##   1. move module's go.mod file to a temporary file
 ##   2  copy main go.mod file to a temporary file to preserve changes
-##   3. run "go mod tidy" to ensure main go.mod file contains all required dependecnies
+##   3. run "go mod tidy" to ensure main go.mod file contains all required dependencies
 ##   4. run controller-gen command
 ##   5. restore go.mod files
 
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	mv ./apis/config/go.mod ./apis/config/go_temp.mod
+	mv ./apis/hoh_heartbeat/go.mod ./apis/hoh_heartbeat/go_temp.mod
```
Comment (vadimeisenbergibm) [inline Makefile:52]: how about using a loop:

`for API in ./apis/*; do mv ./apis/$API/go.mod ./apis/$API/go_temp.mod`?

File: apis/hoh_heartbeat/v1/hoh_heartbeat_types.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HohHeartbeatSpec defines the desired state of HohHeartbeat
+type HohHeartbeatSpec struct{}
```
Comment (vadimeisenbergibm) [inline apis/hoh_heartbeat/v1/hoh_heartbeat_types.go:11]: I think you can just remove the spec field alltogether. Kubernetes CRs do not have to have spec.

**Review by vadimeisenbergibm — CHANGES_REQUESTED** [review]
left some comments

File: Makefile (hunk 1 above)
Comment (OlegSternbergIBM) [reply Makefile]: done

File: apis/hoh_heartbeat/v1/hoh_heartbeat_types.go (hunk 2 above)
Comment (OlegSternbergIBM) [reply apis/hoh_heartbeat/v1/hoh_heartbeat_types.go]: done

**Review by vadimeisenbergibm — APPROVED** [review]

**Pull request merged**